src/compiler/checker.ts

```diff
@@ -14407,8 +14407,8 @@ namespace ts {
             return resultType;
 
             function getTypeAtFlowNode(flow: FlowNode): FlowType {
-                if (flowDepth === 2500) {
-                    // We have made 2500 recursive invocations. To avoid overflowing the call stack we report an error
+                if (flowDepth === 2250) {

```

Comment:
Interesting (i.e. arbitrary :-p ) choice. Seems like if we know 2500 blows the stack in some tests, we might want a little more headroom.

What kind of patterns cause it to get this high? Are there completely valid patterns that require a number up in this range? (i.e. Say hypothetically we set it to 1000. Would there be a noticeably degradation in behavior for a meaningful set of users?)

Reply:
Deeply nested expressions and long statement chains or a combination thereof. Our own codebase gets to around 350 and I've seen some js codebases hit around 1000 before - we should probably try to keep it high, IMO.